Source/WarpX.cpp

```diff
@@ -2574,90 +2467,103 @@ WarpX::AllocLevelMFs (int lev, const BoxArray& ba, const DistributionMapping& dm
-        AllocInitMultiFab(Efield_aux[lev][2], amrex::convert(ba, Ez_nodal_flag), dm, ncomps, ngEB, lev, "Efield_aux[z]", 0.0_rt);
+        m_fields.alloc_init("Efield_aux", Direction{0}, lev, amrex::convert(ba, Ex_nodal_flag), dm, ncomps, ngEB, 0.0_rt);
+        m_fields.alloc_init("Efield_aux", Direction{1}, lev, amrex::convert(ba, Ey_nodal_flag), dm, ncomps, ngEB, 0.0_rt);
+        m_fields.alloc_init("Efield_aux", Direction{2}, lev, amrex::convert(ba, Ez_nodal_flag), dm, ncomps, ngEB, 0.0_rt);
     }
 
     // The external fields that are read from file
     if (m_p_ext_field_params->B_ext_grid_type != ExternalFieldType::default_zero && m_p_ext_field_params->B_ext_grid_type != ExternalFieldType::constant) {
         // These fields will be added directly to the grid, i.e. to fp, and need to match the index type
-        AllocInitMultiFab(Bfield_fp_external[lev][0], amrex::convert(ba, Bfield_fp[lev][0]->ixType()),
-            dm, ncomps, ngEB, lev, "Bfield_fp_external[x]", 0.0_rt);
-        AllocInitMultiFab(Bfield_fp_external[lev][1], amrex::convert(ba, Bfield_fp[lev][1]->ixType()),
-            dm, ncomps, ngEB, lev, "Bfield_fp_external[y]", 0.0_rt);
-        AllocInitMultiFab(Bfield_fp_external[lev][2], amrex::convert(ba, Bfield_fp[lev][2]->ixType()),
-            dm, ncomps, ngEB, lev, "Bfield_fp_external[z]", 0.0_rt);
+        m_fields.alloc_init( "Bfield_fp_external", Direction{0}, lev,
+            amrex::convert(ba, m_fields.get("Bfield_fp",Direction{0},lev)->ixType()),
+            dm, ncomps, ngEB, 0.0_rt);
+        m_fields.alloc_init( "Bfield_fp_external", Direction{1}, lev,
+            amrex::convert(ba, m_fields.get("Bfield_fp",Direction{1},lev)->ixType()),
+            dm, ncomps, ngEB, 0.0_rt);
+        m_fields.alloc_init( "Bfield_fp_external", Direction{2}, lev,
+            amrex::convert(ba, m_fields.get("Bfield_fp",Direction{2},lev)->ixType()),
+            dm, ncomps, ngEB, 0.0_rt);
     }
     if (mypc->m_B_ext_particle_s == "read_from_file") {
         //  These fields will be added to the fields that the particles see, and need to match the index type
-        AllocInitMultiFab(B_external_particle_field[lev][0], amrex::convert(ba, Bfield_aux[lev][0]->ixType()),
-            dm, ncomps, ngEB, lev, "B_external_particle_field[x]", 0.0_rt);
-        AllocInitMultiFab(B_external_particle_field[lev][1], amrex::convert(ba, Bfield_aux[lev][1]->ixType()),
-            dm, ncomps, ngEB, lev, "B_external_particle_field[y]", 0.0_rt);
-        AllocInitMultiFab(B_external_particle_field[lev][2], amrex::convert(ba, Bfield_aux[lev][2]->ixType()),
-            dm, ncomps, ngEB, lev, "B_external_particle_field[z]", 0.0_rt);
+        auto Bfield_aux_levl_0 = m_fields.get("Bfield_aux", Direction{0}, lev);
+        auto Bfield_aux_levl_1 = m_fields.get("Bfield_aux", Direction{1}, lev);
+        auto Bfield_aux_levl_2 = m_fields.get("Bfield_aux", Direction{2}, lev);
```

Comment:
Clang-tidy error "'auto Bfield_aux_levl_0' can be declared as 'auto *Bfield_aux_levl_0' [readability-qualified-auto,-warnings-as-errors]":
```suggestion
        auto *Bfield_aux_levl_0 = m_fields.get("Bfield_aux", Direction{0}, lev);
        auto *Bfield_aux_levl_1 = m_fields.get("Bfield_aux", Direction{1}, lev);
        auto *Bfield_aux_levl_2 = m_fields.get("Bfield_aux", Direction{2}, lev);
```